pyrevolve/pyrevolve.py

```diff
@@ -71,32 +68,26 @@ def __init__(self, checkpoint, fwd_operator, rev_operator,
             raise Exception("Online checkpointing not yet supported. Specify \
                               number of time steps!")
         if(n_checkpoints is None):
-            n_checkpoints = cr.adjust(n_timesteps)
-        if compression_params is None:
-            compression_params = {'scheme': None}
+            self.n_checkpoints = cr.adjust(n_timesteps)
+        else:
+            self.n_checkpoints = n_checkpoints
+
+        if profiler is None:
+            self.profiler = Profiler()
+        else:
+            self.profiler = profiler
+
         self.timings = timings
         self.fwd_operator = fwd_operator
         self.rev_operator = rev_operator
         self.checkpoint = checkpoint
-        compressor, decompressor = init(compression_params)
-        self.profiler = Profiler()
-
-        if compression_params['scheme'] is None:
-            self.storage = NumpyStorage(checkpoint.size, n_checkpoints,
-                                        checkpoint.dtype,
-                                        profiler=self.profiler)
-        else:
-            self.storage = BytesStorage(checkpoint.nbytes, n_checkpoints,
-                                        checkpoint.dtype, auto_pickle=True,
-                                        compression=(compressor, decompressor))
         self.n_timesteps = n_timesteps
-
-        self.scheduler = Revolve(n_checkpoints, n_timesteps)
+        self.scheduler = Revolve(self.n_checkpoints, n_timesteps)
+        self.storage = storage
```

Comment:
Seems to have left a weird corner case where, if the user did not provide a storage, it would be initialised to None? That is not a useful configuration. It would be better to have useful defaults.

Reply:
The idea was to make the BaseRevolver class abstract so that the user can only instantiate either MemoryRevolver or DiskRevolver, but I forgot to make it explicit. In the next commit, all make the BaseRevolver class abstract.